LibraryServices/Services/Library.cs

```diff
@@ -0,0 +1,300 @@
+﻿using System.Collections.Generic;
+using System.Collections;
+using LibraryApi.Models;
+using System.Linq;
+
+namespace LibraryApi.Services
+{
+    public class Library : ILibrary
+    {
+        /// <summary>
+        /// Initialize region
+        /// </summary>
+        #region Initialize
+
+        /// <summary>
+        /// Book's collection
+        /// </summary>
+        private List<Book> _booksInLibrary = new List<Book>();
+
+        /// <summary>
+        /// Author's collection
+        /// </summary>
+        private List<Author> _authors;
+
+        /// <summary>
+        /// Genre's collection
+        /// </summary>
+        private List<Genre> _genres;
+
+        /// <summary>
+        /// Genre's collection
+        /// </summary>
+        private readonly List<KeyValuePair<int, int>> _bookAndGenresConnection;
+
+        /// <summary>
+        /// Service collections initialize
+        /// </summary>
+        public Library(IDataProvider data)
+        {
+            _authors = data.SetAuthors().ToList();
+            _booksInLibrary = data.SetBooks().ToList();
+            _genres = data.SetGenre().ToList();
+            _bookAndGenresConnection = data.SetBookGenreCon().ToList();
+        }
+
+        #endregion
+
+        #region BookRegion
+
+        /// <summary>
+        /// Create new Book in collection
+        /// </summary>
+        /// <param name="book">New Book</param>
+        public void CreateBook(Book book)
+        {
+            _booksInLibrary.Add(book);
+        }
+
+        /// <summary>
+        /// Delete Book by Id
+        /// </summary>
+        /// <param name="id">Book's Id</param>
+        public void DeleteBook(int id)
+        {
+            Book bookToDelete = _booksInLibrary.Find(Book => Book.Id == id);
+            _booksInLibrary.Remove(bookToDelete);
+        }
+
+        /// <summary>
+        /// Return All Books
+        /// </summary>
+        public IEnumerable<Book> GetAllBook()
+        {
+            foreach (Book book in _booksInLibrary)
+            {
+                yield return book;
+            }
+        }
+
+        /// <summary>
+        /// Return Book by Id
+        /// </summary>
+        /// <param name="id">Book's Id</param>
+        /// <returns></returns>
+        public Book GetBook(int id)
+        {
+            return _booksInLibrary.Find(Book => Book.Id == id);
+        }
+
+        /// <summary>
+        /// Update Book in collection
+        /// </summary>
+        /// <param name="id">Book's Id</param>
+        /// <param name="book">New Book info</param>
+        /// <returns>Updated Book</returns>
+        public Book UpdateBook(int id, Book book)
+        {
+            Book bookToUpdate = _booksInLibrary.Find((Book) => Book.Id == id);
+            if (book != null)
```

Comment:
hm if book is null. Probably it's incorrect data for update. Maybe it's a nice idea to throw an exception, handle it and return message to the client. To avoid misunderstanding, the client can think that the update was successful.

Reply:
Also, you should add a unit tests for such negative scenario. As I saw you tested only positive cases